cli/tests/testdata/run/unhandled_rejection_order.ts

```diff
@@ -0,0 +1,3 @@
+for (let i = 0; i < 100; i++) {
+  Promise.reject(i);
+}
```

Comment:
Nice test, very simply reproduction. I would suggest to turn it into a unit test in `core/runtime.rs` instead